src/pybind/mgr/dashboard/controllers/cluster.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+import enum
+
+from .. import mgr
+from . import ApiController, ControllerDoc, Endpoint, EndpointDoc, RESTController
+
+
+class ClusterStatus(enum.Enum):
+    INSTALLED = 0
+    POST_INSTALLED = 1
+
+
+def fetch_status():
+    status = mgr.get_store('cluster-status')
+    if status is None:
+        mgr.set_store('cluster-status', ClusterStatus(0).name)
+        status = mgr.get_store('cluster-status')
+    return status
+
+
+@ApiController('/cluster', secure=False)
+@ControllerDoc("Get Cluster Details", "Cluster")
+class Cluster(RESTController):
+    @Endpoint('GET')
+    @EndpointDoc("Get the cluster status")
+    def get_status(self):
+        return fetch_status()
+
+    @Endpoint('PUT')
+    @EndpointDoc("Update the cluster status",
+                 parameters={'status': (str, 'Cluster Status')})
+    def set_status(self, status: str):
```

Comment:
I got a flake `F811 redefinition of unused ignore` when used the same name for both methods. So I've modified the `set_status` to `update_status` (not sure if it makes any difference) and `get_status` to just `status`

Reply:
The idea of a RESTful API is that you work with entities, concepts (the Resources, which we wrongly refer as endpoints), and the HTTP methods provide the operations (GET, POST, PUT, DELETE, PATCH, HEAD, etc.).

If we have a `Cluster` Resource, the status should be a property of that Resource, and we fetch that with a `GET /cluster` and update that with a `PUT /cluster` (as long as the `status` is the only property of the Resource, otherwise we should use the `PATCH` method).

In any case, for simplicity we are admitting to use `[GET|PUT] /cluster/status` to get fine-grained access to the Resource properties (and because the PUT/PATCH thing is a painful experience).

AFAIK the `@Endpoint()` decorator [allows you to declare the path](https://docs.ceph.com/en/latest/dev/developer_guide/dash-devel/#how-to-add-a-new-controller), so it doesn't need to be linked to the function name.